File: trunk/src/test/java/net/esper/eql/core/TestAliasNodeSwapper.java
package net.esper.eql.core;

import java.util.List;

import net.esper.support.eql.SupportExprNodeFactory;
import net.esper.eql.core.AliasNodeSwapper;
import net.esper.eql.expression.ExprNode;
import net.esper.eql.expression.ExprIdentNode;
import junit.framework.TestCase;

public class TestAliasNodeSwapper extends TestCase 
{
	ExprNode exprTree;
	String alias;
	ExprNode fullExpr;
	ExprNode resultingTree;
	
	public void setUp() throws Exception
	{
		fullExpr = new ExprIdentNode("full expression");
	}
	
	public void testWholeReplaced() throws Exception
	{
		exprTree = new ExprIdentNode("alias");
		alias = "alias";
		resultingTree = AliasNodeSwapper.swap(exprTree, alias, fullExpr);
		assertTrue(resultingTree == fullExpr);
	}
	
	public void testPartReplaced() throws Exception
	{
		exprTree = SupportExprNodeFactory.makeEqualsNode();
		alias = "intPrimitive";
		resultingTree = AliasNodeSwapper.swap(exprTree, alias, fullExpr);
		
		assertTrue(resultingTree == exprTree);
		List<ExprNode> childNodes = resultingTree.getChildNodes();
		List<ExprNode> oldChildNodes = exprTree.getChildNodes();
		assertTrue(childNodes.size() == 2);
		assertTrue(childNodes.get(0) == fullExpr);
		assertTrue(childNodes.get(1) == oldChildNodes.get(1));
		
		exprTree = resultingTree;
		alias = "intBoxed";
		resultingTree = AliasNodeSwapper.swap(exprTree, alias, fullExpr);
		childNodes = resultingTree.getChildNodes();
		assertTrue(childNodes.size() == 2);
		assertTrue(childNodes.get(0) == fullExpr);
		assertTrue(childNodes.get(1) == fullExpr);
		
		exprTree = resultingTree;
		ExprNode newFullExpr = new ExprIdentNode("new full expr");
		alias = "full expression";
		resultingTree = AliasNodeSwapper.swap(exprTree, alias, newFullExpr);
		childNodes = resultingTree.getChildNodes();
		assertTrue(childNodes.size() == 2);
		assertTrue(childNodes.get(0) == newFullExpr);
		assertTrue(childNodes.get(1) == newFullExpr);
	}
	
}

File: NEsper.Tests/eql/core/TestAliasNodeSwapper.cs
using System;
using System.Collections.Generic;

using net.esper.eql.expression;
using net.esper.support.eql;

using NUnit.Core;
using NUnit.Framework;

namespace net.esper.eql.core
{
	
	[TestFixture]
	public class TestAliasNodeSwapper 
	{
		internal ExprNode exprTree;
		internal String alias;
		internal ExprNode fullExpr;
		internal ExprNode resultingTree;
		
		[SetUp]
		public virtual void  setUp()
		{
			fullExpr = new ExprIdentNode("full expression");
		}
		
		[Test]
		public virtual void  testWholeReplaced()
		{
			exprTree = new ExprIdentNode("alias");
			alias = "alias";
			resultingTree = AliasNodeSwapper.Swap(exprTree, alias, fullExpr);
			Assert.IsTrue(resultingTree == fullExpr);
		}
		
		[Test]
		public virtual void  testPartReplaced()
		{
			exprTree = SupportExprNodeFactory.MakeEqualsNode();
			alias = "intPrimitive";
            resultingTree = AliasNodeSwapper.Swap(exprTree, alias, fullExpr);
			
			Assert.IsTrue(resultingTree == exprTree);
			IList<ExprNode> childNodes = resultingTree.ChildNodes;
			IList<ExprNode> oldChildNodes = exprTree.ChildNodes;
			Assert.IsTrue(childNodes.Count == 2);
			Assert.IsTrue(childNodes[0] == fullExpr);
			Assert.IsTrue(childNodes[1] == oldChildNodes[1]);
			
			exprTree = resultingTree;
			alias = "intBoxed";
            resultingTree = AliasNodeSwapper.Swap(exprTree, alias, fullExpr);
			childNodes = resultingTree.ChildNodes;
			Assert.IsTrue(childNodes.Count == 2);
			Assert.IsTrue(childNodes[0] == fullExpr);
			Assert.IsTrue(childNodes[1] == fullExpr);
			
			exprTree = resultingTree;
			ExprNode newFullExpr = new ExprIdentNode("new full expr");
			alias = "full expression";
            resultingTree = AliasNodeSwapper.Swap(exprTree, alias, newFullExpr);
			childNodes = resultingTree.ChildNodes;
			Assert.IsTrue(childNodes.Count == 2);
			Assert.IsTrue(childNodes[0] == newFullExpr);
			Assert.IsTrue(childNodes[1] == newFullExpr);
		}
	}
}
